### PR TITLE
HDS-585: Updated documentation for Focus Colour tokens

### PR DESCRIPTION
**Pull request by minnas**
## Description

This PR adds documentation for focus color tokens. Also updated examples for color tokens 

## Motivation and Context

[HDS-585](https://helsinkisolutionoffice.atlassian.net/browse/HDS-585)

## How Has This Been Tested?

Tested locally

## Demo

[Site Demo](https://city-of-helsinki.github.io/hds-demo/focus-colour-demo-site/foundation/design-tokens/colour/)
[Button Demo](https://city-of-helsinki.github.io/hds-demo/focus-colour-demo-react/?path=/story/components-button--primary)

## TODO

- [ ] fix text to use consistently british english "colour"
- [ ] move focus color to ui colous section
- [ ] improve text so that it doesn't mention the blue (as it can be changed anyway)

[HDS-585]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-585?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: site/src/docs/foundation/design-tokens/colour/index.mdx [hunk 1]
```diff
@@ -33,6 +34,7 @@ The design system uses a colour theme that includes:
 - **Secondary and Tertiary colours:** these provide more ways for accenting select UI elements and distinguishing your service. The secondary and tertiary colours should be applied sparingly and with intent.
 - **UI colours:** the common colours for feedback (error, warning, success, info).
 - **Grayscale:** neutral colour shades.
+- **Focus colors:** the common colours for focus.
```
Comment (mrTuomoK) [inline site/src/docs/foundation/design-tokens/colour/index.mdx:37]: We should either use _color(s)_ or _colour(s)_ in all of our texts, not both 🤓

Comment (mrTuomoK) [reply site/src/docs/foundation/design-tokens/colour/index.mdx:37]: Seems that at least on the code-side it's color so colour(s) -> color(s) is the way to go

Comment (laurakarhu) [reply site/src/docs/foundation/design-tokens/colour/index.mdx:37]: Actually, we have agreed that the documentation in HDS follows British English, so colour is the correct one. In code color is usually special word so it cannot use British typing.

Comment (mrTuomoK) [reply site/src/docs/foundation/design-tokens/colour/index.mdx]: ok... I'm more fond of American english :D :D

**Comment by laurakarhu** [comment]
Todo

- [ ] fix text to use consistently british english "colour"
- [ ] move focus color to ui colous section
- [ ] improve text so that it doesn't mention the blue (as it can be changed anyway)

File: site/src/docs/foundation/design-tokens/colour/index.mdx [hunk 2]
```diff
@@ -76,12 +78,13 @@ Brand colours also have dark, medium light and light variants. You can find thes
...
+| Success                  | Positive state to inform user about success.                                                            | _"Form submit was successful!"_             | <ColorExample color="var(--color-success)" name="Success status colour example" />                     |
+| Alert                    | Warning state to catch users attention but does not require immediate actions.                          | _"Loading is taking longer than expected."_ | <ColorExample color="var(--color-alert)" name="Alert status colour example" />                         |
+| Error                    | Error state which requires users immediate action.                                                      | _"Form is missing critical information."_   | <ColorExample color="var(--color-error)" name="Error status colour example" />                         |
+| Focus                    | Focus colour adds blue outline around the focused element.                                              |                                             | <FocusColorExample color="var(--color-focus-outline)" contentStyle={{height: '28px', width: '28px'}}/> |
```
Comment (harriplappalainen) [inline site/src/docs/foundation/design-tokens/colour/index.mdx:87]: Maybe alter this usage text so it's more in the way of explaining why to use it? Like the states above are explained. Maybe something like "Indication for active selected element which helps users navigating the page with a keyboard."

**Review by harriplappalainen — APPROVED** [review]
That's a good doc 🐕

**Pull request merged**